src/search/search-index/util.js

```diff
@@ -128,6 +128,29 @@ export const reverseRangeLookup = ({ limit = Infinity, ...iteratorOpts }) =>
         })
     })
 
+export const reverseBookmarks = ({ limit = Infinity }) =>
+    new Promise(resolve => {
+        const data = new Map()
+        const stream = index.db.createReadStream({
+            gte: 'bookmark/',
+            lte: 'bookmark/\uffff',
```

Comment:
I think these should be able to be defined dynamically. If you do bookmark search + end time filter, `lte` should be `bookmark/:timestamp` and same with `gte` with start time filter (and combination). That will enable the search area to be reduced when time filters are applied. 

Actually I don't think there's any difference with this function and `reverseRangeLookup`? You should be able to call that with `lte` and `gte` set to `bookmark/*` and it should give the same result